src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -142,13 +238,30 @@ void OTAImageProcessorImpl::HandlePrepareDownload(intptr_t context)
         ChipLogError(SoftwareUpdate, "OTA partition not found");
         return;
     }
+#if CONFIG_ENABLE_DELTA_OTA
+    // New image size is unknown for delta OTA, so we use OTA_SIZE_UNKNOWN flag.
+    esp_err_t err = esp_ota_begin(imageProcessor->mOTAUpdatePartition, OTA_SIZE_UNKNOWN, &(imageProcessor->mOTAUpdateHandle));
+#else
     esp_err_t err =
         esp_ota_begin(imageProcessor->mOTAUpdatePartition, OTA_WITH_SEQUENTIAL_WRITES, &(imageProcessor->mOTAUpdateHandle));
+#endif // CONFIG_ENABLE_DELTA_OTA
+
     if (err != ESP_OK)
     {
         imageProcessor->mDownloader->OnPreparedForDownload(ESP32Utils::MapError(err));
         return;
     }
+#if CONFIG_ENABLE_DELTA_OTA
+    imageProcessor->delta_ota_cfg.user_data               = imageProcessor,
+    imageProcessor->delta_ota_cfg.read_cb                 = &(imageProcessor->DeltaOTAReadCallback),
+    imageProcessor->delta_ota_cfg.write_cb_with_user_data = &(imageProcessor->DeltaOTAWriteCallback),
+
+    imageProcessor->mDeltaOTAUpdateHandle = esp_delta_ota_init(&imageProcessor->delta_ota_cfg);
+    if (imageProcessor->mDeltaOTAUpdateHandle == NULL)
+    {
+        ESP_LOGE(TAG, "esp_delta_ota_init failed");
```

Comment:
delta ota init is failed here, I think we should set the error code using `imageProcessor->mDownloader->OnPreparedForDownload()` and return from here.